### PR TITLE
Create transfer agreement

### PR DESCRIPTION
**Pull request by TomJKing**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala [hunk 1]
```diff
@@ -43,4 +50,34 @@ object Tags {
...
   }
+
+  case class GetConsignmentId(argName: String) extends ValidateTags {
+    override def validate(ctx: Context[ConsignmentApiContext, _]): BeforeFieldResult[ConsignmentApiContext, Unit] = {
+      argName match {
+        case "addTransferAgreementInput" => ctx.arg(argName)
+      }
+    }
+
+  }
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala]: This isn't being used so you can get rid of it.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala [hunk 2]
```diff
@@ -3,7 +3,14 @@ package uk.gov.nationalarchives.tdr.api.graphql
 import sangria.execution.{BeforeFieldResult, FieldTag}
 import sangria.schema.Context
 import uk.gov.nationalarchives.tdr.api.auth.ValidationAuthoriser.{AuthorisationException, continue}
+import uk.gov.nationalarchives.tdr.api.graphql.fields.ConsignmentFields.Consignment
+import uk.gov.nationalarchives.tdr.api.graphql.fields.TransferAgreementFields.AddTransferAgreementInput
 
+import scala.util.{Failure, Success}
+import scala.concurrent._
+import ExecutionContext.Implicits.global
+import scala.concurrent.duration._
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala]: There's a couple of unused imports here.

File: src/test/scala/uk/gov/nationalarchives/tdr/api/routes/TransfersAgreementRouteSpec.scala [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  private def checkTransferAgreementExists(transferAgreementId: Long): Unit = {
+    val sql = s"select * from consignmentapi.TransferAgreement where TransferAgreementId = $transferAgreementId;"
+    val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(sql)
+    val rs: ResultSet = ps.executeQuery()
+    rs.next()
+    rs.getString("TransferAgreementId") should equal(transferAgreementId.toString)
+  }
+
+  private def resetDataBase(): Unit = {
+    DbConnection.db.source.createConnection().prepareStatement("delete from consignmentapi.TransferAgreement").executeUpdate()
```
Comment (MancunianSam) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/routes/TransfersAgreementRouteSpec.scala]: Slightly odd capitalisation

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala (hunk 1 above)
Comment (TomJKing) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala]: Fixed

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala (hunk 2 above)
Comment (TomJKing) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/Tags.scala]: Fixed

File: src/test/scala/uk/gov/nationalarchives/tdr/api/routes/TransfersAgreementRouteSpec.scala (hunk 3 above)
Comment (TomJKing) [reply src/test/scala/uk/gov/nationalarchives/tdr/api/routes/TransfersAgreementRouteSpec.scala]: Fixed

**Pull request merged**